x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_control.tsx

```diff
@@ -96,6 +104,9 @@ export class DrawControl extends Component<Props> {
     this.props.mbMap.getCanvas().style.cursor = '';
     this.props.mbMap.off('draw.modechange', this._onModeChange);
     this.props.mbMap.off('draw.create', this._onDraw);
+    if (this.props.onClick) {
```

Comment:
remove `if (this.props.onClick)` since this._onClick is always defined

Reply:
I ended up surrounding all event subscription logic with `if` statements. I'd rather go in this direction than needlessly subscribe to `click` events then short-circuit it in the callback. It's more lines of code to keep up with possible `undefined` types but I think it's the better option